subgraphs/uniswap-v3/src/common/utils/backfill.ts

```diff
@@ -56,40 +53,70 @@ export function populateEmptyPools(event: ethereum.Event): void {
       "/" +
       token1.name +
       " " +
-      getTradingFee(pool.id).toString() +
+      getTradingFee(poolAddress).toString() +
       "%";
     pool.symbol = token0.name + "/" + token1.name;
     pool.inputTokens = [token0.id, token1.id];
+    pool.inputTokenWeights = [BIGDECIMAL_FIFTY, BIGDECIMAL_FIFTY];
+    pool.tick = BIGINT_ZERO;
     pool.fees = createPoolFees(poolAddress, poolContract.fee());
     pool.isSingleSided = false;
     pool.createdTimestamp = event.block.timestamp;
     pool.createdBlockNumber = event.block.number;
+
+    pool.liquidityToken = null;
+    pool.liquidityTokenType = TokenType.MULTIPLE;
+
     pool.totalValueLockedUSD = BIGDECIMAL_ZERO;
+    pool.totalLiquidity = BIGINT_ZERO;
+    pool.totalLiquidityUSD = BIGDECIMAL_ZERO;
+    pool.activeLiquidity = BIGINT_ZERO;
+    pool.activeLiquidityUSD = BIGDECIMAL_ZERO;
+
+    pool.inputTokenBalances = [BIGINT_ZERO, BIGINT_ZERO];
+    pool.inputTokenBalancesUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+
+    pool.cumulativeTotalVolumeUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeVolumeTokenAmounts = [BIGINT_ZERO, BIGINT_ZERO];
+    pool.cumulativeVolumesUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+
     pool.cumulativeSupplySideRevenueUSD = BIGDECIMAL_ZERO;
     pool.cumulativeProtocolSideRevenueUSD = BIGDECIMAL_ZERO;
     pool.cumulativeTotalRevenueUSD = BIGDECIMAL_ZERO;
-    pool.cumulativeVolumeUSD = BIGDECIMAL_ZERO;
-    pool.inputTokenBalances = [BIGINT_ZERO, BIGINT_ZERO];
-    pool.inputTokenWeights = [BIGDECIMAL_FIFTY, BIGDECIMAL_FIFTY];
+
+    pool.uncollectedProtocolSideTokenAmounts = [BIGINT_ZERO, BIGINT_ZERO];
+    pool.uncollectedProtocolSideValuesUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+    pool.uncollectedSupplySideTokenAmounts = [BIGINT_ZERO, BIGINT_ZERO];
+    pool.uncollectedSupplySideValuesUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+
+    pool.positionCount = INT_ZERO;
+    pool.openPositionCount = INT_ZERO;
+    pool.closedPositionCount = INT_ZERO;
+
+    pool._totalAmountWithdrawn = [BIGINT_ZERO, BIGINT_ZERO];
+    pool._totalAmountCollected = [BIGINT_ZERO, BIGINT_ZERO];
+    pool._totalAmountEarned = [BIGINT_ZERO, BIGINT_ZERO];
+    pool._mostRecentSnapshotsDayID = INT_ZERO;
+    pool._mostRecentSnapshotsHourID = INT_ZERO;
```

Comment:
Is this the only place where you are setting the default 0 values for a pool? If not, can you combine them all into one? If not, I fear that whenever a new field is added for the Pool entity we’ll forget to add it here too.

Reply:
Agreed. Fixed.